sources/github.go

```diff
@@ -652,6 +657,30 @@ func (g *githubSource) GetDefaultBranch(ctx context.Context, accessToken *Access
 	return *gitRepo.DefaultBranch, nil
 }
 
+func (g *githubSource) waitForCommit(ctx context.Context, accessToken *AccessToken, owner, repo, sha string) error {
+	githubClient := g.interactionsFunc(ctx, accessToken.Token, accessToken.Type, g.cfg.RateLimitTimeoutSeconds, g.cfg.RateLimitRetryCount)
+
+	err := retry.Retry(time.Duration(g.cfg.WaitTagTimeoutSeconds)*time.Second, func(i int) error {
+		ref, _, err := githubClient.GetRepoRef(ctx, owner, repo, "heads/main")
+		if err != nil {
+			return err
+		}
+
+		commit, err := githubClient.GetCommit(ctx, owner, repo, *ref.Object.SHA)
+		if err != nil {
+			return err
+		}
+
+		if *commit.SHA != sha {
+			return errors.Wrapf(ErrCommitNotFound, "last commit is not %s", sha)
+		}
+
+		return nil
```

Comment:
Could this be conflated into a simple `GetCommit` with the desired SHA? or do we need to check that this is the top commit?